src/runtime.js

```diff
@@ -889,3 +895,48 @@ export function registerElementForTesting(win, elementName) {
   win.AMP.registerElement(element.name, element.implementationClass,
       element.css);
 }
+
+/**
+ * For a given extension, checks that its version is the same
+ * as the version of the main AMP binary.
+ * If yes, returns false and does nothing else.
+ * If they are different, returns false, and initiates a load
+ * of the respective extension via a versioned URL.
+ *
+ * This is currently guarded by the 'version-locking' experiment.
+ * With this active, all scripts in a given page are guaranteed
+ * to have the same AMP release version.
+ *
+ * @param {!Window} win
+ * @param {function(!Object)|ExtensionPayloadDef} fnOrStruct
+ * @return {boolean}
+ */
+function maybeLoadCorrectVersion(win, fnOrStruct) {
+  if (!isExperimentOn(win, 'version-locking')) {
+    return false;
+  }
+  if (typeof fnOrStruct == 'function') {
```

Comment:
Bunch of code that wants AMP to initialize does, not necessarily extensions. E.g. viewer integration scripts (e.g. from Google Search) do that and bunch of PWA functions.